Source/Charts/Renderers/LegendRenderer.swift

```diff
@@ -59,6 +59,7 @@ open class LegendRenderer: NSObject, Renderer
                         entries.append(
                             LegendEntry(
                                 label: sLabels[j % sLabels.count],
+                                labelColor: nil,
```

Comment:
is it possible to use default value? seeing `labelColor: nil` repeat

Reply:
We could but the rest of that function doesn’t. Seems weird to have a default at the start of the function. That’s my preference though

Reply:
We could but the rest of that function doesn’t. Seems weird to have a default at the start of the function. That’s my preference though

Reply:
ok. if @jjatie has no question or no response for a while, we could merge then.

Reply:
Cool. Trying to finish up 4.0

Reply:
I think the issue is that we are putting too many things in the initializer. The only thing we need to create a valid `LegendEntry` is the `label`, everything else is optional. The `label` should be a constant and the rest of the properties remain vars.

Reply:
It might even make sense to only allow the entry group to be created through the empty initializer, i.e.:
```
@objc(ChartLegendEntry)
open class LegendEntry: NSObject
{
@objc(ChartLegendEntry)
open class LegendEntry: NSObject
{
    public override init()
    {
        super.init()
    }
    
    /// A label to start a group.
    @objc public init(label: String?)
    {
        self.label = label
    }
    public override init()
    {
        super.init()
    }
    
    /// - parameter label:                  The legend entry text.
    @objc public init(label: String)
    {
        self.label = label
    }
    
    // ...
}
```

Reply:
Don’t know if the empty initializer is useful but I agree with your points. I’ll change some stuff